flink-streaming-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -341,23 +355,49 @@ public void setInputType(TypeInformation<?> type, ExecutionConfig executionConfi
 	}
 
 	@Override
-	public void open(Configuration parameters) throws Exception {
-		super.open(parameters);
+	public void initializeState(FunctionInitializationContext context) throws Exception {
+		Preconditions.checkArgument(this.restoredBucketStates == null, "The operator has already been initialized.");
 
-		subtaskIndex = getRuntimeContext().getIndexOfThisSubtask();
+		initFileSystem();
 
-		state = new State<T>();
```

Comment:
undoing it was a bit of an overstatement; moving it into a separate commit with all the other formatting changes would do the trick as well. If you try that out you will see that the changes to open and initializeState no longer intersect so heavily.